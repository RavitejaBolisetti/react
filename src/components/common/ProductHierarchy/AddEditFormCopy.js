import React, { useEffect,useState } from 'react';
import { Col, Input, Form, Row, Select, Switch,Drawer,Button } from 'antd';
// import { FaSearch } from 'react-icons/fa';
import { validateRequiredInputField, validateRequiredSelectField, validationFieldLetterAndNumber } from 'utils/validation';

import styles from 'pages/common/Common.module.css';
import TreeSelectField from '../TreeSelectField';
import { preparePlaceholderSelect, preparePlaceholderText } from 'utils/preparePlaceholder';

import style from 'components/common/DrawerAndTable.module.css';


const { Option } = Select;
const { TextArea } = Input;

const AddEditFormMain = ({ open,form,setDrawer,setFormBtnDisable,handleUpdate2,footerEdit,setsaveclick,isChecked, setIsChecked, handleAttributeChange, setSelectedTreeKey, setSelectedTreeSelectKey, flatternData, formActionType, fieldNames, isReadOnly, formData, selectedTreeKey, selectedTreeSelectKey, isDataAttributeLoaded, attributeData, setIsModalOpen, setFieldValue, handleSelectTreeClick, productHierarchyData,isLoadingOnSave }) => {
    const treeFieldNames = { ...fieldNames, label: fieldNames.title, value: fieldNames.key };
    const [isChildAllowed, setIsChildAllowed] = useState(true);
    const [isVisible,setIsVisible] = useState(false)

    const disabledProps = { disabled: isReadOnly };

    let treeCodeId = '';
    let treeCodeReadOnly = false;

    if (formActionType === 'edit' || formActionType === 'view') {
        treeCodeId = formData?.parntProdctId;
    } else if (formActionType === 'child') {
        treeCodeId = selectedTreeKey && selectedTreeKey[0];
        treeCodeReadOnly = true;
    } else if (formActionType === 'sibling') {
        treeCodeReadOnly = true;
        const treeCodeData = flatternData.find((i) => selectedTreeKey[0] === i.key);
        treeCodeId = treeCodeData && treeCodeData?.data?.parntProdctId;
    }

    let drawerTitle = 'Add Product Details';
    if (formActionType === 'add') {
        drawerTitle = 'Add Product Details';
    } else if (formActionType === 'update') {
        drawerTitle = 'Edit Product Details';
    } else if (formActionType === 'view') {
        drawerTitle = 'View Product Details';
    }

    const handleForm = () => {
        setFormBtnDisable(true);
    };

    
    const onClose = () => {
        setDrawer(false);
        setFormBtnDisable(false);
        form.
            resetFields();
    };


    useEffect(() => {
        if (formActionType === 'sibling') {
            setSelectedTreeKey([treeCodeId]);
        }

        setSelectedTreeSelectKey(treeCodeId);
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [treeCodeId]);

    const treeSelectFieldProps = {
        treeFieldNames,
        treeData: productHierarchyData,
        treeDisabled: treeCodeReadOnly || isReadOnly,
        selectedTreeSelectKey,
        handleSelectTreeClick,
        defaultValue: treeCodeId,
        placeholder: preparePlaceholderSelect('Parent'),
    };

    const onCloseAction = () => {
        console.log("hello")
    }
    // const isVisible = () => {
    //     console.log("hello")
    // }
    const onOpenAction = () => {
        console.log("hello")
    }
    useEffect(()=>{
        const selectedAttribute = attributeData?.find((i) => i.id === formData?.attributeKey);  
        setIsChildAllowed(selectedAttribute?.hierarchyAttribueName)
    })
    return (
        <>
            <Drawer
                title={drawerTitle}
                placement="right"
                onClose={onClose}
                className={footerEdit ? style.viewMode : style.drawerCriticalityGrp}
                width="540px"
                open={open}
                footer={
                    <>
                        <Row gutter={20}>
                            <Col xs={8} sm={8} md={8} lg={8} xl={8} xxl={8}>
                                <Button danger onClick={onClose} className={style.cancelBtn}>
                                    Cancel
                                </Button>
                            </Col>
                            <Col xs={16} sm={16} md={16} lg={16} xl={16} xxl={16} className={style.saveBtn}>
                                <Button loading={isLoadingOnSave} onClick={() => setsaveclick(true)} form="myForm" key="submit" htmlType="submit" type="primary">
                                    Save
                                </Button>

                                {footerEdit ? (
                                    <Button onClick={handleUpdate2} form="myForm" key="submitAndNew" htmlType="submit" type="primary">
                                        Edit
                                    </Button>
                                ) : (
                                    ''
                                )}
                            </Col>
                        </Row>
                    </>
                }
            >
                <>
                    <Row gutter={20}>
                        <Col xs={24} sm={24} md={24} lg={24} xl={24}>
                            <Form.Item initialValue={formData?.attributeKey} name="attributeKey" label="Attribute Level" rules={[validateRequiredSelectField('Geographical Attribute Level')]}>
                                <Select onChange={handleAttributeChange} loading={!isDataAttributeLoaded} placeholder={preparePlaceholderSelect('Attribute Level')} {...disabledProps} showSearch allowClear>
                                    {attributeData?.map((item) => (
                                        <Option value={item?.id}>{item?.hierarchyAttribueName}</Option>
                                    ))}
                                </Select>
                            </Form.Item>
                        </Col>
                    </Row>
                    <Row>

                        <Col xs={24} sm={24} md={24} lg={24} xl={24} className={styles.padRight18}>
                            <Form.Item initialValue={treeCodeId} label="Parent" name="parntProdctId">
                                <TreeSelectField {...treeSelectFieldProps} />
                            </Form.Item>
                        </Col>
                    </Row>

                    <Row gutter={20}>
                        <Col xs={24} sm={24} md={24} lg={24} xl={24}>
                            <Form.Item label="Code" name="prodctCode" initialValue={formData?.prodctCode} rules={[validateRequiredInputField('Code'), validationFieldLetterAndNumber('Code')]}>
                                <Input placeholder={preparePlaceholderText('Code')} maxLength={6} className={styles.inputBox} disabled={formData?.id || isReadOnly} />
                            </Form.Item>
                        </Col>
                    </Row>
                    <Row>
                        <Col xs={24} sm={24} md={24} lg={24} xl={24}>
                            <Form.Item name="prodctShrtName" label="Short Description" initialValue={formData?.prodctShrtName} rules={[validateRequiredInputField('Short Description')]}>
                                <Input className={styles.inputBox} placeholder={preparePlaceholderText('Short Description')} {...disabledProps} />
                            </Form.Item>
                        </Col>
                    </Row>
                    <Row gutter={20}>
                        <Col xs={24} sm={24} md={24} lg={24} xl={24}>
                            <Form.Item name="prodctLongName" label="Long Description" initialValue={formData?.prodctLongName} rules={[validateRequiredInputField('Long Description')]}>
                                <TextArea rows={1} placeholder={preparePlaceholderText('Long Description')} {...disabledProps} />
                            </Form.Item>
                        </Col>
                    </Row>
                    <Row>
                        <Col xs={24} sm={24} md={24} lg={24} xl={24} className={styles.padLeft10}>
                            <Form.Item initialValue={formData?.active === 'Y' ? 1 : 0} label="Status" name="active">
                                <Switch value={formData?.active === 'Y' ? 1 : 0} checkedChildren="Active" unCheckedChildren="Inactive" defaultChecked {...disabledProps} />
                            </Form.Item>
                        </Col>
                    </Row>
                </>
            </Drawer>
        </>
    );
};

export const AddEditForm =(AddEditFormMain);
