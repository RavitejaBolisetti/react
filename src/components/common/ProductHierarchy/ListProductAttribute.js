import React from 'react';
import { Fragment } from 'react';

import CardItemProductAttribute from './CardItemProductAttribute';

const ListProductAttribute = (props) => {
    const { form, skuAttributes } = props;
    return (
        <>
            {skuAttributes.length > 0 &&
                skuAttributes.map((action) => {
                    return <CardItemProductAttribute {...action} form={form} />;
                })}
        </>
    );
};

export default ListProductAttribute;
