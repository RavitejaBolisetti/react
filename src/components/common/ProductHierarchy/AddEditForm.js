import React, { useEffect, useState } from 'react';
import { Col, Input, Form, Row, Collapse, Select, Switch, Button } from 'antd';
import { withDrawer } from 'components/withDrawer';
import style from '../../common/DrawerAndTable.module.css';
import { PlusBorderedIcon, MinusBorderedIcon } from 'Icons';
import ProductAttributeAddEditForm from './ProductAttributeDetail/ProductAttributeAddEditForm';
import ListProductAttribute from './ProductAttributeDetail/ListProductAttribute';

import styles from 'components/common/Common.module.css';
import TreeSelectField from '../TreeSelectField';
import { FROM_ACTION_TYPE } from 'constants/formActionType';

import { validateRequiredInputField, validateRequiredSelectField, validationFieldLetterAndNumber,validateAlphanumericWithSpaceHyphenPeriod } from 'utils/validation';
import { preparePlaceholderSelect, preparePlaceholderText } from 'utils/preparePlaceholder';

const { Option } = Select;
const { TextArea } = Input;
const { Panel } = Collapse;

const AddEditFormMain = (props) => {
    const { onCloseAction, handleAttributeChange, formActionType, fieldNames, isReadOnly = false, formData, isDataAttributeLoaded, attributeData, productHierarchyData } = props;
    const { selectedTreeKey, setSelectedTreeKey, selectedTreeSelectKey, setSelectedTreeSelectKey, handleSelectTreeClick, flatternData } = props;
    const { isFormBtnActive, setFormBtnActive } = props;
    const { form, skuAttributes, setSKUAttributes } = props;

    console.log('formData', formData);
    const [actionForm] = Form.useForm();
    const [openAccordian, setOpenAccordian] = useState(1);
    const [isAddBtnDisabled, setAddBtnDisabled] = useState(false);
    const [showProductAttribute, setShowProductAttribute] = useState(true);

    const { onFinish, onFinishFailed } = props;

    const treeFieldNames = { ...fieldNames, label: fieldNames?.title, value: fieldNames?.key };

    const disabledProps = { disabled: isReadOnly };

    let treeCodeId = '';
    let treeCodeReadOnly = false;

    if (formActionType === FROM_ACTION_TYPE.EDIT) {
        treeCodeId = formData?.parntProdctId;
    } else if (formActionType === FROM_ACTION_TYPE.CHILD) {
        treeCodeId = selectedTreeKey && selectedTreeKey[0];
        treeCodeReadOnly = true;
    } else if (formActionType === FROM_ACTION_TYPE.SIBLING) {
        treeCodeReadOnly = true;
        const treeCodeData = flatternData.find((i) => selectedTreeKey[0] === i.key);
        treeCodeId = treeCodeData && treeCodeData?.data?.parntProdctId;
    }

    useEffect(() => {
        setSelectedTreeSelectKey(treeCodeId);
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [treeCodeId]);

    const treeSelectFieldProps = {
        treeFieldNames,
        treeData: productHierarchyData,
        treeDisabled: treeCodeReadOnly || isReadOnly,
        selectedTreeSelectKey,
        handleSelectTreeClick,
        defaultValue: treeCodeId,
        placeholder: preparePlaceholderSelect('parent'),
    };

    const handleFormValueChange = () => {
        setFormBtnActive(true);
    };

    const handleFormFieldChange = () => {
        setFormBtnActive(true);
    };
    const handleCollapse = (key) => {
        setOpenAccordian((prev) => (prev === key ? '' : key));
    };

    const handleProductchange = (e) => {
        const value = e.target.textContent;
        setShowProductAttribute(value);
    };

    const onActionFormFinish = (val) => {
        const { value, label } = val?.attributeName;
        setSKUAttributes((prev) => [...prev, { attributeName: label, id: value, attributeValue: val.attributeValue }]);
        actionForm.resetFields();
    };

    const attributeFormProps = {
        form,
        skuAttributes: formData?.skuAttributes,
        setSKUAttributes,
        isAddBtnDisabled,
        setAddBtnDisabled,
        onFinish: onActionFormFinish,
    };

    return (
        <>
            <Form autoComplete="off" form={form} layout="vertical" onValuesChange={handleFormValueChange} onFieldsChange={handleFormFieldChange} onFinish={onFinish} onFinishFailed={onFinishFailed}>
                <Row gutter={20}>
                    <Col xs={24} sm={24} md={24} lg={24} xl={24}>
                        <Form.Item initialValue={formData?.attributeKey} name="attributeKey" label="Attribute Level" rules={[validateRequiredSelectField('attribute level')]}>
                            <Select onChange={handleAttributeChange} onClick={handleProductchange} loading={!isDataAttributeLoaded} placeholder={preparePlaceholderSelect('attribute level')} {...disabledProps} showSearch allowClear>
                                {attributeData?.map((item) => (
                                    <Option value={item?.id}>{item?.hierarchyAttribueName}</Option>
                                ))}
                            </Select>
                        </Form.Item>
                    </Col>

                    <Col xs={24} sm={24} md={24} lg={24} xl={24} className={styles.padRight18}>
                        <Form.Item initialValue={treeCodeId} label="Parent" name="parntProdctId">
                            <TreeSelectField {...treeSelectFieldProps} />
                        </Form.Item>
                    </Col>
                </Row>

                <Row gutter={20}>
                    <Col xs={24} sm={24} md={24} lg={24} xl={24}>
                        <Form.Item label="Code" name="prodctCode" initialValue={formData?.prodctCode} rules={[validateRequiredInputField('code'), validationFieldLetterAndNumber('code')]}>
                            <Input placeholder={preparePlaceholderText('code')} maxLength={6} className={styles.inputBox} disabled={formData?.id || isReadOnly} />
                        </Form.Item>
                    </Col>

                    <Col xs={24} sm={24} md={24} lg={24} xl={24}>
                        <Form.Item name="prodctShrtName" label="Short Description" initialValue={formData?.prodctShrtName} rules={[validateRequiredInputField('short description'),validateAlphanumericWithSpaceHyphenPeriod('short description')]}>
                            <Input className={styles.inputBox} placeholder={preparePlaceholderText('short description')} maxLength={50} {...disabledProps} />
                        </Form.Item>
                    </Col>
                </Row>

                <Row gutter={20}>
                    <Col xs={24} sm={24} md={24} lg={24} xl={24}>
                        <Form.Item name="prodctLongName" label="Long Description" initialValue={formData?.prodctLongName} rules={[validateRequiredInputField('long description'),validateAlphanumericWithSpaceHyphenPeriod('long description')]}>
                            <TextArea rows={2} placeholder={preparePlaceholderText('long description')} showCount maxLength={100} {...disabledProps} />
                        </Form.Item>
                    </Col>

                    <Col xs={24} sm={24} md={24} lg={24} xl={24} className={styles.padLeft10}>
                        <Form.Item initialValue={formActionType === 'child' || formActionType === 'sibling' ? true : formData?.active ? true : false} label="Status" name="active">
                            <Switch value={formActionType === 'child' || formActionType === 'sibling' ? true : formData?.active ? true : false} checkedChildren="Active" unCheckedChildren="Inactive" defaultChecked={formActionType === 'child' || formActionType === 'sibling' ? true : formData?.active === true || null || undefined ? true : false} {...disabledProps} />
                        </Form.Item>
                    </Col>
                </Row>

                <Row gutter={20} className={styles.formFooter}>
                    <Col xs={12} sm={12} md={12} lg={12} xl={12} className={styles.footerBtnLeft}>
                        <Button danger onClick={onCloseAction}>
                            Cancel
                        </Button>
                    </Col>

                    <Col xs={12} sm={12} md={12} lg={12} xl={12} className={styles.footerBtnRight}>
                        <Button htmlType="submit" danger type="primary" disabled={!isFormBtnActive}>
                            Save
                        </Button>
                    </Col>
                </Row>

                {showProductAttribute ? (
                    <Collapse className={openAccordian === 1 ? style.accordianHeader : ''} onChange={() => handleCollapse(1)} expandIcon={({ isActive }) => (isActive ? <MinusBorderedIcon /> : <PlusBorderedIcon />)}>
                        <Panel header={<span className={openAccordian === 1 ? style.accordianHeader : ''}>Product Atrribute Details</span>} key="1">
                            <ProductAttributeAddEditForm {...attributeFormProps} />
                            <ListProductAttribute {...attributeFormProps} />
                        </Panel>
                    </Collapse>
                ) : null}
            </Form>
        </>
    );
};

export const AddEditForm = withDrawer(AddEditFormMain, {});
