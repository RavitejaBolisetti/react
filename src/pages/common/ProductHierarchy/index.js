import { ProductHierarchy } from './ProductHierarchy';

export { ProductHierarchy };
