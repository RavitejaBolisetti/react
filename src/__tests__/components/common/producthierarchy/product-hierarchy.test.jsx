import '@testing-library/jest-dom/extend-expect';
import customRender from '@utils/test-utils';
import { ProductHierarchy } from '@components/common/ProductHierarchy/ProductHierarchy';
import { fireEvent, screen } from '@testing-library/react';
import createMockStore from '__mocks__/store';
import { Provider } from 'react-redux';

afterEach(() => {
    jest.restoreAllMocks();
});

describe('Producthierarchy Components', () => {
    it('should render producthierarchy component', () => {
        customRender(<ProductHierarchy />);
    });

    it('search input box should work', () => {
        const mockStore = createMockStore({
            auth: { userId: 106 },
            data: {
                ProductHierarchy: {
                    organizationId: true,
                },
            },
        });

        customRender(
            <Provider store={mockStore}>
                <ProductHierarchy />
            </Provider>
        );
        const searchBox = screen.getByRole('textbox', { name: '' });
        fireEvent.change(searchBox, { target: { value: 'Kai' } });
    });

    it('add and edit form cancel button should work', async () => {
        const mockStore = createMockStore({
            auth: { userId: 106 },
            data: {
                ProductHierarchy: {
                    organizationId: true,
                },
            },
        });

        customRender(
            <Provider store={mockStore}>
                <ProductHierarchy onCloseAction={jest.fn()} />
            </Provider>
        );
        const addBtn = screen.getByRole('button', { name: 'plus Add' });

        fireEvent.click(addBtn);

        const cancelBtn = screen.getByRole('button', { name: 'Cancel' });

        fireEvent.click(cancelBtn);
    });

    it('Hierarchy tree select should work', async () => {
        const mockStore = createMockStore({
            auth: { userId: 106 },
            data: {
                ProductHierarchy: {
                    organizationId: true,
                    data: [{ id: '106', prodctLongName: 'Kai', prodctShrtName: 'Kai', prodctCode: '106', active: true, attributeKey: '106' }],
                    changeHistoryVisible: true,
                    isVisible: true,
                },
                HierarchyAttributeMaster: {
                    data: [{ status: 'active' }],
                },
            },
        });

        customRender(
            <Provider store={mockStore}>
                <ProductHierarchy onCloseAction={jest.fn()} fetchProductDetail={jest.fn()} />
            </Provider>
        );
        const treeSelect = screen.getByText('Kai');

        fireEvent.click(treeSelect);
    });

    it('Hierarchy tree select should work with selecting tree', async () => {
        const mockStore = createMockStore({
            auth: { userId: 106 },
            data: {
                ProductHierarchy: {
                    organizationId: false,
                },
                ManufacturerOrgHierarchy: {
                    data: [{ id: '106', manufactureOrgLongName: 'Kai', manufactureOrgShrtName: 'Kai', manufactureOrgCode: 'Kai', manufactureOrgParntId: 'null', active: true, attributeKey: '106', subManufactureOrg: [] }],
                },
            },
        });

        customRender(
            <Provider store={mockStore}>
                <ProductHierarchy onCloseAction={jest.fn()} />
            </Provider>
        );
        const treeSelect = screen.getByRole('combobox', { name: '' });
        fireEvent.change(treeSelect, { target: { value: 'Kai' } });
        const treeData = screen.getByText('Kai');

        fireEvent.click(treeData);
    });
});
