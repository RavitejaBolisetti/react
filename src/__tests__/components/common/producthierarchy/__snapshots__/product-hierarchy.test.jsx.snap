// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Producthierarchy Components should render Producthierarchy components 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: overlay;"
  >
    <div>
      <div
        class="contentHeaderBackground"
        data-testid="form-data"
      >
        <div
          class="ant-row css-dev-only-do-not-override-14wwjjs"
          style="margin-left: -10px; margin-right: -10px;"
        >
          <div
            class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-18 ant-col-lg-18 ant-col-xl-18 css-dev-only-do-not-override-14wwjjs"
            style="padding-left: 10px; padding-right: 10px;"
          >
            <form
              autocomplete="off"
              class="ant-form ant-form-horizontal css-dev-only-do-not-override-14wwjjs masterListSearchForm"
            >
              <div
                class="ant-form-item css-dev-only-do-not-override-14wwjjs"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-14wwjjs"
                >
                  <div
                    class="ant-col ant-form-item-label css-dev-only-do-not-override-14wwjjs"
                  >
                    <label
                      class="ant-form-item-no-colon"
                      for="code"
                      title="Hierarchy"
                    >
                      Hierarchy
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-14wwjjs"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <div
                          class="ant-row css-dev-only-do-not-override-14wwjjs"
                          id="code"
                          style="margin-left: -10px; margin-right: -10px;"
                        >
                          <div
                            class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-12 ant-col-lg-12 ant-col-xl-12 css-dev-only-do-not-override-14wwjjs"
                            data-testid="treeSelectField"
                            style="padding-left: 10px; padding-right: 10px;"
                          >
                            <div
                              class="ant-select ant-tree-select ant-select-in-form-item css-dev-only-do-not-override-14wwjjs ant-select-single ant-select-allow-clear ant-select-show-arrow ant-select-show-search"
                              style="width: 100%;"
                            >
                              <div
                                class="ant-select-selector"
                              >
                                <span
                                  class="ant-select-selection-search"
                                >
                                  <input
                                    aria-autocomplete="list"
                                    aria-controls="rc_select_TEST_OR_SSR_list"
                                    aria-expanded="false"
                                    aria-haspopup="listbox"
                                    aria-owns="rc_select_TEST_OR_SSR_list"
                                    autocomplete="off"
                                    class="ant-select-selection-search-input"
                                    id="rc_select_TEST_OR_SSR"
                                    role="combobox"
                                    type="search"
                                    value=""
                                  />
                                </span>
                                <span
                                  class="ant-select-selection-placeholder"
                                >
                                  Select organization hierarchy
                                </span>
                              </div>
                              <span
                                aria-hidden="true"
                                class="ant-select-arrow"
                                style="user-select: none;"
                                unselectable="on"
                              >
                                <span
                                  aria-label="down"
                                  class="anticon anticon-down ant-select-suffix"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="down"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
      <div
        class="ant-row css-dev-only-do-not-override-14wwjjs"
        span="24"
        style="margin-left: -10px; margin-right: -10px;"
      >
        <div
          class="ant-col borderBottomCorner productHierarchy marT20 ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24 css-dev-only-do-not-override-14wwjjs"
          style="padding-left: 10px; padding-right: 10px;"
        >
          <div
            class="content"
          >
            <div
              class="emptyContainer"
            >
              <div
                class="css-dev-only-do-not-override-14wwjjs ant-empty ant-empty-normal"
              >
                <div
                  class="ant-empty-image"
                  style="height: 60px;"
                >
                  <svg
                    height="41"
                    viewBox="0 0 64 41"
                    width="64"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                      transform="translate(0 1)"
                    >
                      <ellipse
                        cx="32"
                        cy="33"
                        fill="#f5f5f5"
                        rx="32"
                        ry="7"
                      />
                      <g
                        fill-rule="nonzero"
                        stroke="#d9d9d9"
                      >
                        <path
                          d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                        />
                        <path
                          d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                          fill="#fafafa"
                        />
                      </g>
                    </g>
                  </svg>
                </div>
                <div
                  class="ant-empty-description"
                >
                  <span
                    class="descriptionText"
                  >
                    Please select hierarchy type to view records.
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      ;
    </div>
  </body>,
  "container": <div>
    <div
      class="contentHeaderBackground"
      data-testid="form-data"
    >
      <div
        class="ant-row css-dev-only-do-not-override-14wwjjs"
        style="margin-left: -10px; margin-right: -10px;"
      >
        <div
          class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-18 ant-col-lg-18 ant-col-xl-18 css-dev-only-do-not-override-14wwjjs"
          style="padding-left: 10px; padding-right: 10px;"
        >
          <form
            autocomplete="off"
            class="ant-form ant-form-horizontal css-dev-only-do-not-override-14wwjjs masterListSearchForm"
          >
            <div
              class="ant-form-item css-dev-only-do-not-override-14wwjjs"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-14wwjjs"
              >
                <div
                  class="ant-col ant-form-item-label css-dev-only-do-not-override-14wwjjs"
                >
                  <label
                    class="ant-form-item-no-colon"
                    for="code"
                    title="Hierarchy"
                  >
                    Hierarchy
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-14wwjjs"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <div
                        class="ant-row css-dev-only-do-not-override-14wwjjs"
                        id="code"
                        style="margin-left: -10px; margin-right: -10px;"
                      >
                        <div
                          class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-12 ant-col-lg-12 ant-col-xl-12 css-dev-only-do-not-override-14wwjjs"
                          data-testid="treeSelectField"
                          style="padding-left: 10px; padding-right: 10px;"
                        >
                          <div
                            class="ant-select ant-tree-select ant-select-in-form-item css-dev-only-do-not-override-14wwjjs ant-select-single ant-select-allow-clear ant-select-show-arrow ant-select-show-search"
                            style="width: 100%;"
                          >
                            <div
                              class="ant-select-selector"
                            >
                              <span
                                class="ant-select-selection-search"
                              >
                                <input
                                  aria-autocomplete="list"
                                  aria-controls="rc_select_TEST_OR_SSR_list"
                                  aria-expanded="false"
                                  aria-haspopup="listbox"
                                  aria-owns="rc_select_TEST_OR_SSR_list"
                                  autocomplete="off"
                                  class="ant-select-selection-search-input"
                                  id="rc_select_TEST_OR_SSR"
                                  role="combobox"
                                  type="search"
                                  value=""
                                />
                              </span>
                              <span
                                class="ant-select-selection-placeholder"
                              >
                                Select organization hierarchy
                              </span>
                            </div>
                            <span
                              aria-hidden="true"
                              class="ant-select-arrow"
                              style="user-select: none;"
                              unselectable="on"
                            >
                              <span
                                aria-label="down"
                                class="anticon anticon-down ant-select-suffix"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
    <div
      class="ant-row css-dev-only-do-not-override-14wwjjs"
      span="24"
      style="margin-left: -10px; margin-right: -10px;"
    >
      <div
        class="ant-col borderBottomCorner productHierarchy marT20 ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24 css-dev-only-do-not-override-14wwjjs"
        style="padding-left: 10px; padding-right: 10px;"
      >
        <div
          class="content"
        >
          <div
            class="emptyContainer"
          >
            <div
              class="css-dev-only-do-not-override-14wwjjs ant-empty ant-empty-normal"
            >
              <div
                class="ant-empty-image"
                style="height: 60px;"
              >
                <svg
                  height="41"
                  viewBox="0 0 64 41"
                  width="64"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="none"
                    fill-rule="evenodd"
                    transform="translate(0 1)"
                  >
                    <ellipse
                      cx="32"
                      cy="33"
                      fill="#f5f5f5"
                      rx="32"
                      ry="7"
                    />
                    <g
                      fill-rule="nonzero"
                      stroke="#d9d9d9"
                    >
                      <path
                        d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                      />
                      <path
                        d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                        fill="#fafafa"
                      />
                    </g>
                  </g>
                </svg>
              </div>
              <div
                class="ant-empty-description"
              >
                <span
                  class="descriptionText"
                >
                  Please select hierarchy type to view records.
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    ;
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
